### PR TITLE
Python 3: Update StringIO()

### PR DESCRIPTION
**Pull request by luckyh**
Signed-off-by: Xu Han <xuhan@redhat.com>

### CODE REVIEW

File: virttest/asset.py [hunk 1]
```diff
@@ -4,8 +4,8 @@
 import string
 import types
 import glob
-import StringIO
```
Comment (chunfuwen) [inline virttest/asset.py]: Just reference: avocado almost remove StringIO,use BytesIO instead, see https://github.com/avocado-framework/avocado/pull/2441

Comment (chunfuwen) [inline virttest/asset.py]: Just reference: avocado almost remove StringIO,use BytesIO instead, see https://github.com/avocado-framework/avocado/pull/2441

File: virttest/asset.py [hunk 2]
```diff
@@ -64,7 +64,7 @@ def __init__(self, cfg, tmpdir=data_dir.get_tmp_dir(), raise_errors=False):
             # Config file is just a string, convert it to a python file like
             # object using StringIO
             else:
-                self.cfg = StringIO.StringIO(cfg)
+                self.cfg = StringIO(cfg)
```
Comment (chunfuwen) [inline virttest/asset.py]: I observed some difference result on python2 and python3:
'''
[root@chunfulaptop avocado-vt]# python
Python 2.7.5 (default, May  3 2017, 07:55:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-14)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from io import StringIO
>>> file = StringIO('/home/chwem/git/avocado-vt/shared/cfg/base.cfg')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: initial_value must be unicode or None, not str
>>> exit()
[root@chunfulaptop avocado-vt]# python3
Python 3.4.5 (default, Dec 11 2017, 14:22:24) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from io import StringIO
>>> file = StringIO('/home/chwem/git/avocado-vt/shared/cfg/base.cfg')
>>> 

'''

Comment (chunfuwen) [inline virttest/asset.py]: On python2 , I try import StringIO and from io import StringIO ,but get different result as follow:
```
[root@chunfulaptop avocado-vt]# python
Python 2.7.5 (default, May  3 2017, 07:55:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-14)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from io import StringIO
>>> f = StringIO('hello')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: initial_value must be unicode or None, not str
>>>

[root@chunfulaptop avocado-vt]# python
Python 2.7.5 (default, May  3 2017, 07:55:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-14)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import StringIO
>>> f = StringIO.StringIO('hello')
>>> f.getvalue()
'hello'
>>> 

```

**Comment by chunfuwen** [comment]
@luckyh 
@hsmj1412  
Sound great, I try this solution on python 2 and 3 as below:
```
[root@chunfulaptop chwem]# python2
Python 2.7.5 (default, May  3 2017, 07:55:04) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-14)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> try:
...     from StringIO import StringIO
... except ImportError:
...     from io import StringIO
... 
>>> value = StringIO('helloworld')
>>> value.getvalue()
'helloworld'
>>> exit()
[root@chunfulaptop chwem]# python3
Python 3.4.5 (default, Dec 11 2017, 14:22:24) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> try:
...     from StringIO import StringIO
... except ImportError:
...     from io import StringIO
... 
>>> value = StringIO('helloworld')
>>> value.getvalue()
'helloworld'
>>> 

```

**Comment by luckyh** [comment]
@chunfuwen @hsmj1412 updated, thanks.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Pull request merged**